src/Pkg.jl

```diff
@@ -312,7 +312,9 @@ const build = API.build
     Pkg.pin(pkgs::Union{PackageSpec, Vector{PackageSpec}}; io::IO=stderr, all_pkgs::Bool=false)
 
 Pin a package to the current version (or the one given in the `PackageSpec`) or to a certain
-git revision. A pinned package is never updated. To pin all dependencies set `all_pkgs=true`.
+git revision. A pinned package is never automatically updated: if `pkg` is tracking a path,
+or a repository, those remain tracked in a frozen state). For example, if one wishes to update
+a pinned package in the current manifest, one must explicitly call `Pkg.pin` again.
```

Comment:
Does you have to call `Pkg.unpin` before `Pkg.update("Foo")`?

Sorry for the questions, I just don't use pin on tracked repos like this

Reply:
Yes, in my experiments calling `Pkg.add` and `Pkg.update` on a pinned package is a noop.

One must either `Pkg.unpin` and update or one can also call `Pkg.pin` again, in order to update the manifest when `Project.toml` has changed (this is my use case: I track a path which is dev'ed, and pinned).

This is what I expect from pinning a package: freeze it until I explicitly want to work on it.

Reply:
Ok perhaps `one must unpin, update then pin the package again`

Reply:
Thanks, (hopefully) done in https://github.com/JuliaLang/Pkg.jl/pull/3148/commits/414f9a0f40d098c13a7f1eb1ce70700feecddf2e.